### PR TITLE
[Merged by Bors] - feat: drop an assumption in `mfderivWithin_const`

### PR DESCRIPTION
**Pull request by sgouezel**
Followup to #19694 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Geometry/Manifold/MFDeriv/SpecificFunctions.lean [hunk 1]
```diff
@@ -198,9 +198,10 @@ theorem mfderiv_const :
     mfderiv I I' (fun _ : M => c) x = (0 : TangentSpace I x →L[𝕜] TangentSpace I' c) :=
   HasMFDerivAt.mfderiv (hasMFDerivAt_const c x)
 
-theorem mfderivWithin_const (hxs : UniqueMDiffWithinAt I s x) :
-    mfderivWithin I I' (fun _ : M => c) s x = (0 : TangentSpace I x →L[𝕜] TangentSpace I' c) :=
-  (hasMFDerivWithinAt_const _ _ _).mfderivWithin hxs
+theorem mfderivWithin_const :
+    mfderivWithin I I' (fun _ : M => c) s x = (0 : TangentSpace I x →L[𝕜] TangentSpace I' c) := by
+  apply HasMFDerivWithinAt.mfderivWithin_eq_zero
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/MFDeriv/SpecificFunctions.lean:203]: Up to you if you want to golf this back to one line.

File: Mathlib/Geometry/Manifold/MFDeriv/Basic.lean [hunk 2]
```diff
@@ -693,11 +693,24 @@ protected theorem MDifferentiableAt.mfderiv (h : MDifferentiableAt I I' f x) :
...
 
-theorem HasMFDerivWithinAt.mfderivWithin (h : HasMFDerivWithinAt I I' f s x f')
+protected theorem HasMFDerivWithinAt.mfderivWithin (h : HasMFDerivWithinAt I I' f s x f')
     (hxs : UniqueMDiffWithinAt I s x) : mfderivWithin I I' f s x = f' := by
   ext
   rw [hxs.eq h h.mdifferentiableWithinAt.hasMFDerivWithinAt]
 
+theorem HasMFDerivWithinAt.mfderivWithin_eq_zero (h : HasMFDerivWithinAt I I' f s x 0) :
+    mfderivWithin I I' f s x = 0 := by
+  simp only [mfderivWithin, h.mdifferentiableWithinAt, ↓reduceIte, writtenInExtChartAt, extChartAt,
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/MFDeriv/Basic.lean:703]: This is a (bit of a) pain to read: is there a way to make this prettier? For instance, can you use `mfld_simps` to golf?

**Review by grunweg — COMMENTED** [review]
Thanks! Two optional suggestions for improvements.
maintainer delegate

**Review by jcommelin — COMMENTED** [review]
bors d+

**Pull request closed without merging**